docs/explanations/context-aware-policies.md

```diff
@@ -54,27 +52,29 @@ spec:
   mutating: false
 ```
 
-Policy authors can provide the list of Kubernetes resources accessed by their context aware policy by annotating the policy file. Kubewarden administrators can look into the policy metadata using the `kwctl inspect` command and obtain a list of Kubernetes resources the policy requires access to. This list can then be used to populate the `ClusterAdmissionPolicy` definition.
+Once deployed, this policy can read the data of the `deployment` and `pod` resources.
+
+Policy authors provide lists of Kubernetes resources for their context aware policy.
+This is done by annotating the policy.
+Kubewarden administrators view policy metadata using the `kwctl inspect` command.
+They can get a list of resources the policy needs access to.
+An administrator uses this list to populate the `ClusterAdmissionPolicy` definition.
 
 :::danger
-Kubewarden administrators must review carefully the kind of resources the policy is going to access
-to prevent system abuses.
+To prevent system abuse, Kubewarden administrators must review the resources the policy will access.
 
-For example, a policy evaluating Ingress objects would have very good reasons to read all the `Ingress`
-resources already defined inside of the cluster. However it would be hard to justify why the same policy
-would also require access to `Secret` resources.
+For example, a policy evaluating Ingress objects would have good reasons to read the `Ingress` resources defined in the cluster.
+The same policy can't justify having access to `Secret` resources.
 
-When reviewing the list of resources a context aware policy wants to have access to, approach this process with the same mindset used when evaluating the privileges granted to a smart phone App you're about to install on your mobile device.
-Would you install a torch App that requires access to your address book or your location?
+Policies should have the least access needed to function correctly.
 
-While policies have read-only access to Kubernetes resources, a malicious attacker might abuse a Kubewarden policy to exfiltrate sensitive data from the cluster.
 :::
 
+<!--TODO: What is this note for? what does it mean? Is it relevant to context aware policies?-->
```

Comment:
Yes, this note is relevant to the context aware policies.

Some background, all the Kubernetes resources are identified via `apiVersion` and `kind`. Usually `apiVersion` is a string following this pattern: `<api>/<version>`. For example, the Deployment objects have `apps/v1` as `apiVersion`, while Ingress have `networking.k8s.io/v1`. However, all the resources that belong to the `core` group (like Pod, Service, Node, ConfigMap, Secret,...) have a different pattern: `<version>`.

This is relevant for context aware policies because the admin must define the allow list of resources the policy can read. 

Having this rule would not work (meaning the policy would not be granted access to Pod):

```yaml  contextAwareResources:
    - apiVersion: "core/v1"
      kind: "pod"
```

While this one would:

```yaml  contextAwareResources:
    - apiVersion: "v1"
      kind: "pod"
```

BTW, most Kubernetes users should not be surprised by that syntax give this is what is used when defining a Pod, Service,... via a yaml file.

Reply:
Thanks Flavio.

Reply:
Hi @flavio I've made changes to the note at the end. Hopefully it is technical accurate, clear and readable. Thanks.